src/QsCompiler/Tests.Compiler/LocalVerificationTests.fs

```diff
@@ -261,6 +261,9 @@ type LocalVerificationTests () =
         this.Expect "UsingRenamedAttribute1"       [Warning WarningCode.DeprecationWithRedirect]
         this.Expect "UsingRenamedAttribute2"       [Warning WarningCode.DeprecationWithRedirect]
         this.Expect "UsingRenamedAttribute3"       [Warning WarningCode.DeprecationWithRedirect]
+
+        this.Expect "UsingNestedDeprecatedCallable"[Warning WarningCode.DeprecationWithRedirect]
+        //this.Expect "UsingDepAttrInDepCall"        [Warning WarningCode.DeprecationWithoutRedirect]
```

Comment:
I thought NestedDeprecatedCallable is supposed to generate a "nested" warning-since I marked it as @deprecated"nested" (the one inside is a "" warning), and when I use it is should only give the outer warning? Why should it give no warning?

Reply:
I'm actually a little confused as of how to test one warning instead of two

Reply:
`NestedDeprecatedCallable` shouldn't generate a warning, because even though it calls a deprecated callable, `NestedDeprecatedCallable` is itself deprecated, so the warning should be suppressed. It's not a warning to declare a callable as deprecated with the `@Deprecated` attribute - just when you use a callable that has the attribute.

To test that there's only one warning generated for each pair of functions, you can use `this.Expect "Foo" []` to assert that the compiler diagnostics for the first function in the pair is the empty list (no warnings).

Reply:
Got it - so simply doing "DeprecatedCallable" will not generate any warning either.

Reply:
The compiler parses everything, even if it's not used. We just have two rules for `@Deprecated`:

1. When a callable is *declared* with `@Deprecated`, it doesn't generate a warning, but anything that *calls* it will generate a warning.
2. Rule 1 doesn't apply if the call is made from inside another callable that was declared with `@Deprecated`. (That's the rule that this PR is adding.)

I may have misunderstood your question, let me know if that answers it.